lib/permutations.rb

```diff
@@ -1,4 +1,34 @@
-
+# returns true if one string is a permutation ("hello", "ehllo")of the other
 def permutations?(string1, string2)
+    if string1.length == 0 && string2.length == 0
+        return true
+    end
   
+    string1_hash = {}
+    string1_arr = string1.split("")
+
+    string1_arr.each do | char |
+        if string1_hash[char]
+            string1_hash[char] += 1
+        else
+            string1_hash[char] = 1
+        end
+    end
+
+    i = 0
+    while i < string2.length
+        char = string2[i]
+        if string1_hash[char]
```

Comment:
You need an `else` for this if to return `false` if the hash doesn't have `char`.